torch/_dynamo/output_graph.py

```diff
@@ -941,6 +968,21 @@ def create_proxy(
         type_expr=None,
         proxy_factory_fn=None,
     ):
+        # If there are any freevars, lift them to being inputs
+        if self.parent is not None:
+            flat_args, _ = pytree.tree_flatten(args)
+            for arg in flat_args:
+                if not isinstance(arg, torch.fx.Proxy):
+                    # Is a constant
+                    continue
+                if arg in self.seen_proxies:
+                    continue
+                if not hasattr(arg, "node"):
+                    continue
+                if arg.node.name in self.input_name_to_proxy:
+                    continue
+                self.lift_freevar_to_input(arg)
```

Comment:
Don't I want to do this even if I don't have a parent?

Reply:
It already gets taken care of if you don't have a parent. If Dynamo sees an untracked Tensor in LOAD_GLOBAL or LOAD_DEREF, it will begin tracking it, and as a part of that it will lift the freevar to the graph input by calling `tx.output.create_graph_input`.

Maybe I should rename `lift_freevar_to_input` to `lift_tracked_freevar_to_input`

Reply:
ack - this is fine, but add a comment :P